src/components/pages/Notes/NoteDialog.js

```diff
@@ -1,65 +1,66 @@
-import React, { useState, useRef, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Modal, Button, Form } from 'react-bootstrap';
-import useUser from '../../../contexts/UserContext';
-import PropTypes from 'prop-types';
+import useNotes from '../../../contexts/NotesContext';
 import Moment from 'react-moment';
 
-const NoteDialog = ({ dialogVisible, closeDialog, note }) => {
-  const [show, setShow] = useState(dialogVisible);
+const NoteDialog = () => {
+  const [, dispatch, { dialogVisible, closeDialog, note }] = useNotes();
+  const [formData, setFormData] = useState({ title: '', content: '' });
+  const { title, content } = formData;
 
-  const didMountRef = useRef(false);
   useEffect(() => {
-    if (didMountRef.current) {
-      setShow(dialogVisible);
-    } else didMountRef.current = true;
-
-    if (note) {
-      setFormData({ title: note.title, content: note.content });
-    }
-  }, [setShow, dialogVisible, note]);
-
-  const onClose = e => closeDialog && closeDialog(e);
-
-  const [, dispatch] = useUser();
-  const [formData, setFormData] = useState({
-    title: '',
-    content: ''
-  });
+    dialogVisible && note && setFormData(() => ({ ...note }));
+  }, [dialogVisible, note]);
 
-  const { title, content } = formData;
-
-  const onChange = e => setFormData({ ...formData, [e.target.name]: e.target.value });
+  const onChange = e => {
+    e.target && setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
 
   const onSubmit = e => {
     e.preventDefault();
     if (!note) {
-      if (title !== '' && content !== '') {
-        dispatch({
-          type: 'ADD_NOTE',
-          payload: { content }
-        });
+      if (!title) {
+        return; // TODO: add invalid note error
```

Comment:
It's not really needed here because action checks for title existence as well. There's no need to verify that twice.